src/aiortc/contrib/media.py

```diff
@@ -258,33 +326,49 @@ def player_worker(
-                np.save(os.path.join(save_dir, 'sender_frame_%05d.npy' % frame.index), 
+                #pass
+                np.save(os.path.join(save_dir, 'sender_frame_%05d.npy' % frame.index),
                         frame_array)
 
-            # Put in a separate track from which keypoints will be extracted
             if enable_prediction:
                 logger.warning(
                     "MediaPlayer(%s) Put frame %s in keypoints queue: %s",
                      container.name, str(frame.index), str(frame)
                 )
-                asyncio.run_coroutine_threadsafe(keypoints_track._queue.put((frame_array, frame_time, \
-                        frame.index, frame.pts)), loop)
+ 
+                if prediction_type != "keypoints":
+                    #TODO: using the interpolate function will result in stall of receiver in the pipeline
+                    '''
+                    driving = frame_to_tensor(img_as_float32(frame_array), device)
+                    driving_lr = F.interpolate(driving, lr_size).data.cpu().numpy()
+                    driving_lr = np.transpose(driving_lr, [0, 2, 3, 1])[0]
+                    driving_lr *= 255
+                    driving_lr = driving_lr.astype(np.uint8)
+
+                    lr_frame = av.VideoFrame.from_ndarray(driving_lr)
+                    lr_frame.pts = frame.pts
+                    lr_frame.time_base = frame.time_base
+                    '''
+                    lr_frame = frame.reformat(width=lr_size, height=lr_size)
+                    place_frame_in_video_queue(lr_frame, loop, lr_video_track, container)
+                    if save_lr_video_npy and save_dir is not None:
+                        lr_frame_array = lr_frame.to_rgb().to_ndarray()
+                        np.save(os.path.join(save_dir,
+                                'sender_lr_frame_%05d.npy' % lr_frame.index),
+                                lr_frame_array)
+
+                else:
+                    asyncio.run_coroutine_threadsafe(keypoints_track._queue.put((frame_array, frame_time, \
+                            frame.index, frame.pts)), loop)
 
             # Only add video frame is this is meant to be used as a source \
             # frame or if prediction is disabled
             if (enable_prediction and frame.index % reference_update_freq == 0) or \
-                    not enable_prediction:
-                frame_time = frame.time
-                frame_index = frame.index
-                frame = stamp_frame(frame, frame.index, frame.pts, frame.time_base)
-                
-                logger.warning(
-                    "MediaPlayer(%s) Put video frame %s in the queue: %s",
-                     container.name, str(frame_index), str(frame)
-                )
-                asyncio.run_coroutine_threadsafe(video_track._queue.put((frame, frame_time, \
-                        frame_index, frame.pts)), loop)
+                    display_option == 'real' or not enable_prediction:
```

Comment:
where is this display_option controlled from right now?

Reply:
In the future hopefully from the feedback that the controller gives